framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2783,6 +2795,17 @@ public ParameterizedExecutableType constructorFromUse(NewClassTree tree) {
             // Reset the enclosing type because it can be substituted incorrectly.
             ((AnnotatedDeclaredType) con.getReturnType()).setEnclosingType(enclosingType);
         }
+
+        // Adapt parameters, which makes parameters and arguments be the same size for later
+        // checking, always update varargsType and parameter types together.
+        AnnotatedArrayType varargsType = null;
+        List<AnnotatedTypeMirror> parameters;
+        if (con.getElement().isVarArgs()) {
+            parameters = con.getParameterTypes();
+            varargsType = (AnnotatedArrayType) parameters.get(parameters.size() - 1);
+        }
+        parameters = AnnotatedTypes.adaptParameters(this, con, tree.getArguments());
+        con.setParameterTypes(parameters, varargsType);
```

Comment:
This part is the same for both `NewClassTree` and `MethodInvocationTree`, do you want to extract it into a private function?